src/AppBundle/Controller/CategoryController.php

```diff
@@ -31,8 +31,26 @@ public function categoryDetail(Request $request)
 			throw new NotFoundHttpException("Kategorie neexistuje");
 		}
 
+		// paginator
+        $em = $this->get('doctrine.orm.entity_manager');
+        $dql = 'SELECT p
```

Comment:
Tento blok DQL kódu je takmer totožný s ProductRepository->findByCategory() s tým rozdielom, že KnpPaginatorBundle to vyžaduje bez getResult() /takže som tu nemohol priamo použiť túto metódu/.
Vznikla mi tu však duplicita, tak by ma zaujímalo, či je nejaký best practice ako to presunúť do ProductRepository s tým aby som mal k dispozícií metódu bez a zároveň s getResult(). Samozrejme, napadajú ma možnosti ako to je možné spraviť, ale skôr ma zaujíma to, čo je osvedčené z praxe.
Vďaka.